crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -542,8 +542,16 @@ impl RoomEventCacheInner {
         // Add all the events to the backend.
         trace!("adding new events");
 
-        let (stored_prev_batch_token, timeline_event_diffs) =
-            self.state.write().await.handle_sync(timeline).await?;
+        let (stored_prev_batch_token, timeline_event_diffs) = self
+            .state
+            .write()
+            .await
+            .handle_sync(
+                timeline,
+                #[cfg(feature = "experimental-search")]
+                &self.weak_room.get().expect("Should be able to get room"),
```

Comment:
This `.expect()` is wrong, I think; it could be that the client is shutting down. Instead, let's do `let Some(room) = self.weak_room.get() else { warn!("..."); return; }`